python/ray/data/_internal/execution/resource_manager.py

```diff
@@ -126,6 +144,50 @@ def _estimate_object_store_memory(self, op, state) -> int:
 
         return mem_op_internal + mem_op_outputs
 
+    def _estimate_object_store_memory_per_node(
+        self, op: "PhysicalOperator", state: "OpState"
+    ) -> Dict[str, int]:
+        usage_per_node = defaultdict(int)
+        # Don't count input refs towards dynamic memory usage, as they have been
+        # pre-created already outside this execution.
+        if isinstance(op, InputDataBuffer):
+            return usage_per_node
```

Comment:
Can you help me understand why there's this special-case?

Reply:
Result of `_estimate_object_store_memory` is used to calculate the budget of each op.
We don't want to count InputDataBuffer's usage towards the current pipeline.